### PR TITLE
Action cls export

### PR DESCRIPTION
**Pull request by jaegukhyun**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algo/action_classification/mmconfigs/x3d.yaml [hunk 1]
```diff
@@ -15,14 +15,15 @@ cls_head:
 data_preprocessor:
   format_shape: NCTHW
   mean:
-    - 114.75
-    - 114.75
-    - 114.75
+    - 0.0
```
Comment (vinnamkim) [inline src/otx/algo/action_classification/mmconfigs/x3d.yaml]: Are these corrected by the pretrained checkpoint?

File: src/otx/algo/action_classification/openvino_model.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    return x
+
+
+def get_multiclass_predictions(logits: np.ndarray, activate: bool = False) -> ClassificationResult:
+    """Get multiclass predictions."""
+    index = np.argmax(logits)
+    if activate:
+        logits = softmax_numpy(logits)
+
+    return ClassificationResult([(index, index, logits[index])], np.ndarray(0), np.ndarray(0), np.ndarray(0))
```
Comment (vinnamkim) [inline src/otx/algo/action_classification/openvino_model.py]: Why don't you create an output sink for softmax at model exportation?

File: src/otx/core/data/transform_libs/torchvision.py [hunk 3]
```diff
@@ -125,6 +128,159 @@ def _get_preprocess_shape(self, oldh: int, oldw: int, long_side_length: int) ->
         return (newh, neww)
 
 
+class DecodeVideo(tvt_v2.Transform):
```
Comment (vinnamkim) [inline src/otx/core/data/transform_libs/torchvision.py:131]: I'd rather move the decode process to `Dataset._get_item() -> tv_tensor.Video`, not in the transform pipeline. Can you do this work at the next PR?

File: src/otx/core/model/entity/action_classification.py [hunk 4]
```diff
@@ -39,6 +60,7 @@ def __init__(self, num_classes: int, config: DictConfig) -> None:
         config = inplace_num_classes(cfg=config, num_classes=num_classes)
         self.config = config
         self.load_from = config.pop("load_from", None)
+        self.image_size = (1, 1, 3, 8, 224, 224)
```
Comment (vinnamkim) [inline src/otx/core/model/entity/action_classification.py:63]: Why are there six dimensions? Is it `NCTHW`?

File: src/otx/core/model/entity/base.py [hunk 5]
```diff
@@ -448,3 +449,8 @@ def _read_ptq_config_from_ir(self, ov_model: Model) -> dict[str, Any]:
         initial_ptq_config = argparser.parse_object(initial_ptq_config)
 
         return argparser.instantiate_classes(initial_ptq_config).as_dict()
+
+    @property
+    def model_parameters(self) -> dict:
```
Comment (sovrasov) [inline src/otx/core/model/entity/base.py]: I'd rename this to something related to model adapter extra parameters, since this is not related to export

File: src/otx/recipe/_base_/data/mmaction_base.yaml [hunk 6]
```diff
@@ -5,7 +5,7 @@ config:
   mem_cache_img_max_size:
     - 500
     - 500
-  image_color_channel: RGB
+  image_color_channel: BGR
```
Comment (sovrasov) [inline src/otx/recipe/_base_/data/mmaction_base.yaml]: Why do we have to use BGR? Is it because the pretrained weights expect BGR?

File: src/otx/recipe/_base_/data/mmaction_base.yaml [hunk 7]
```diff
@@ -78,6 +85,8 @@ config:
         scale:
           - -1
           - 256
+      - type: CenterCrop
```
Comment (sovrasov) [inline src/otx/recipe/_base_/data/mmaction_base.yaml:88]: Is that center crop reproduced in OV action model?

File: src/otx/algo/action_classification/mmconfigs/x3d.yaml (hunk 1 above)
Comment (jaegukhyun) [reply src/otx/algo/action_classification/mmconfigs/x3d.yaml]: I just sync with previous OTX1.x configs. I'll modify after fixing performance issues in action classification

File: src/otx/algo/action_classification/openvino_model.py (hunk 2 above)
Comment (jaegukhyun) [reply src/otx/algo/action_classification/openvino_model.py]: Actually, softmax is already performed by model. That's the reason why default value of activate is False. This is for unexpected situation which model do not calculate softmax on its final output

File: src/otx/core/data/transform_libs/torchvision.py (hunk 3 above)
Comment (jaegukhyun) [reply src/otx/core/data/transform_libs/torchvision.py:131]: Sure I'll take a look at this on next PR

File: src/otx/core/model/entity/action_classification.py (hunk 4 above)
Comment (jaegukhyun) [reply src/otx/core/model/entity/action_classification.py:63]: I don't know why but mmaction requires another dimension in front of batch.

File: src/otx/recipe/_base_/data/mmaction_base.yaml (hunk 6 above)
Comment (jaegukhyun) [reply src/otx/recipe/_base_/data/mmaction_base.yaml]: Yes

File: src/otx/recipe/_base_/data/mmaction_base.yaml (hunk 7 above)
Comment (jaegukhyun) [reply src/otx/recipe/_base_/data/mmaction_base.yaml:88]: Nope, but it is for sync with previous OTX1.x pipeline. I'll revisit this pipeline after fixing some performance issue in action classification. Currently there's accuracy drop in OTX2.0

File: src/otx/core/model/entity/base.py (hunk 5 above)
Comment (jaegukhyun) [reply src/otx/core/model/entity/base.py]: 32eee609db5e3d5ed958bcf6487d6179ac308c33 Done.

File: src/otx/core/model/entity/action_classification.py (hunk 4 above)
Comment (vinnamkim) [reply src/otx/core/model/entity/action_classification.py:63]: Can you find a root cause of this and fix in the next PR?

Comment (jaegukhyun) [reply src/otx/core/model/entity/action_classification.py:63]: I'm not sure, mmaction do
image
during feature extraction
image
It is used train, predict, export

File: src/otx/algo/action_classification/mmconfigs/x3d.yaml (hunk 1 above)
Comment (vinnamkim) [reply src/otx/algo/action_classification/mmconfigs/x3d.yaml]: You must align it with the environment where the pretraining checkpoint was created since we we finetune from it.
I can see that it takes RGB and different channel normalization as opposed to those you changed
https://github.com/open-mmlab/mmaction2/blob/4d6c93474730cad2f25e51109adcf96824efc7a3/configs/recognition/x3d/metafile.yml#L17
https://github.com/open-mmlab/mmaction2/blob/4d6c93474730cad2f25e51109adcf96824efc7a3/configs/recognition/x3d/x3d_m_16x5x1_facebook-kinetics400-rgb.py#L1
https://github.com/open-mmlab/mmaction2/blob/4d6c93474730cad2f25e51109adcf96824efc7a3/configs/_base_/models/x3d.py#L15-L16

File: src/otx/algo/action_classification/openvino_model.py (hunk 2 above)
Comment (vinnamkim) [reply src/otx/algo/action_classification/openvino_model.py]: We assume that this class only allows the model exported by OTX. From your explanation, there is no case that the model output is not probability.

File: src/otx/recipe/_base_/data/mmaction_base.yaml (hunk 6 above)
Comment (jaegukhyun) [reply src/otx/recipe/_base_/data/mmaction_base.yaml]: I changed it to RGB [792ac01](https://github.com/openvinotoolkit/training_extensions/pull/2961/commits/792ac01e4abf151f20667610dfb840783cab365d)

File: src/otx/algo/action_classification/mmconfigs/x3d.yaml (hunk 1 above)
Comment (jaegukhyun) [reply src/otx/algo/action_classification/mmconfigs/x3d.yaml]: Done [792ac01](https://github.com/openvinotoolkit/training_extensions/pull/2961/commits/792ac01e4abf151f20667610dfb840783cab365d)

File: src/otx/algo/action_classification/openvino_model.py (hunk 2 above)
Comment (jaegukhyun) [reply src/otx/algo/action_classification/openvino_model.py]: 8a90193 Done

File: src/otx/core/model/entity/action_classification.py (hunk 4 above)
Comment (vinnamkim) [reply src/otx/core/model/entity/action_classification.py:63]: I can see that the default test pipeline uses `ThreeCrop`
https://github.com/open-mmlab/mmaction2/blob/4d6c93474730cad2f25e51109adcf96824efc7a3/configs/recognition/x3d/x3d_m_16x5x1_facebook-kinetics400-rgb.py#L18
If so, isn't it correct that it should be `(1,3,3,8,224,224)`?

Comment (jaegukhyun) [reply src/otx/core/model/entity/action_classification.py:63]: image
OTX does CenterCrop

Comment (vinnamkim) [reply src/otx/core/model/entity/action_classification.py:63]: Did you evaluate on the side effect about that mismatch?

Comment (jaegukhyun) [reply src/otx/core/model/entity/action_classification.py:63]: I will during fixing performance issue in OTX2.0. Actually OTX1.x show good performance with previous pipelines

**Review by vinnamkim — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]
LGTM

File: src/otx/recipe/_base_/data/mmaction_base.yaml (hunk 6 above)
Comment (sovrasov) [reply src/otx/recipe/_base_/data/mmaction_base.yaml]: > I changed it to RGB [792ac01](https://github.com/openvinotoolkit/training_extensions/pull/2961/commits/792ac01e4abf151f20667610dfb840783cab365d)

I see, anyway, it's up to you, Maybe RGB would hurt the final accuracy, if the weights expect BGR

**Review by sovrasov — APPROVED** [review]
